internal/controller/appconfigurationprovider_controller.go

```diff
@@ -250,6 +250,22 @@ func (reconciler *AzureAppConfigurationProviderReconciler) Reconcile(ctx context
 
 	/* Create secret when there are secret settings */
 	if processor.RefreshOptions.SecretSettingPopulated {
+		// Verify the existence of the secret which is not owned by the current provider
+		for name := range processor.Settings.SecretSettings {
```

Comment:
aslo I don't understand the code at https://github.com/Azure/AppConfiguration-KubernetesProvider/blob/main/internal/controller/processor.go#L223C2-L228C3.

```
	for secretName, targetSecret := range resolvedSecretData {
		existingSecret, ok := existingSecrets[secretName]
		if ok {
			maps.Copy(existingSecret.Data, targetSecret.Data)
		}
	}
```

Reply:
That's because we only resolve the non-versioned secret, so we need to merge the new values into the existing secret to update the non-versioned secrets

Reply:
after checking go lang doc, I know why it confused me. `maps.Copy` actually copies data from the second parameter to the first, not the other way around. The parameter `targetSecret` makes me think the second is the destination. Can we rename it to `resolvedSecret`?

Reply:
Updated

Reply:
thanks